slides/_naming_for_clarity.qmd

```diff
@@ -18,4 +18,53 @@ A few naming tips and conventions:
 
 ## Naming For Clarity
 
-Some examples:
\ No newline at end of file
+Some examples:
+
+- Pronounceable names without abbreviations: \
+ `ms`, `chclt`, `stm` --> `mass`, `chocolate`, `stem`
+- Naming for the content, not the type: \
+`array`, `list`, `age_int`, `country_set` --> `dogs`, `cats`, `age`, `countries`
+- Naming booleans: \
+`purple`, `not_plant`, `sidekick` --> `is_purple`, `is_plant`, `has_sidekick`
+
+## Explaining Variables
+
+Without explaining variables, it is hard to see what this code is doing:
+
+```
+import re
+
+re.match("^\\+?[1-9][0-9]{7,14}$", "+12223334444")
+```
+
+## With explaining variables:
+
+It is easier to see the intention. The code is more self-documenting.
+
+```
+import re
+
+validate_phone_number_pattern = "^\\+?[1-9][0-9]{7,14}$"
+re.match(validate_phone_number_pattern, "+12223334444")
+```
+
+## Exercise 3 {.smaller}
+
+Look through the code for any names of methods or variables that could be improved or
+clarified and update them. Note if you are using an IDE like Intellij or VSCode,
+you can use automatic renaming.
+Can you find an example from each of the suggestions listed below?
+Does this make the code easier to follow?
+
+Consider the following:
+
+- The name should show the intention, think about how someone else might read it (this could be future you)
+- Use pronounceable names e.g. `mass` not `ms`, `stem` not `stm`
+- avoid abbreviations and single letter variable names where possible
+- One word per concept e.g. choose one of `put`, `insert`, `add` in the same code base
+- Use names that can be searched
+- Describe content rather than storage type
+- Naming booleans, use prefixes like `is`, `has` or `can` and avoid negations like `not_green`
+- Plurals to indicate groups, e.g. a list of dog objects would be `dogs`, not `dog_list`
+- Keep it simple and use technical terms where appropriate
+- Use explaining variables
```

Comment:
This list runs off of the end of the slide when I render them and view.
See similar comment above for a possible solution.